core/dotnet/src/serialization/JsonSerializationWriter.cs

```diff
@@ -25,29 +27,41 @@ public JsonSerializationWriter()
                 writer.WriteStringValue(value);
             }
         }
-        public void WriteBoolValue(string key, bool value) {
-            if(!string.IsNullOrEmpty(key)) writer.WritePropertyName(key);
-            writer.WriteBooleanValue(value);
+        public void WriteBoolValue(string key, bool? value) {
+            if(!string.IsNullOrEmpty(key) && value.HasValue) writer.WritePropertyName(key);
+            if(value.HasValue) writer.WriteBooleanValue(value.Value);
```

Comment:
yes as `false` is valid json. We (graph) wrap everything in objects but I believe that's mostly because of OData, from a JSON point of view, you don't need to. Also, I'm using the method for arrays, to write the values of the array (without keys)